From 32642c65e31c26673cc4687eefacdf0087cf4498 Mon Sep 17 00:00:00 2001
From: Carlos Rafael Giani <dv@pseudoterminal.org>
Date: Mon, 28 Jul 2014 00:10:03 +0200
Subject: [PATCH] Remove X libraries from GYP files

Some X libraries are linked even when ozone-wayland is used, causing
linker errors if no X libraries are around

Upstream-Status: Pending

Signed-off-by: Carlos Rafael Giani <dv@pseudoterminal.org>
---
 third_party/libjingle/source/talk/libjingle.gyp                 | 4 ----
 third_party/libjingle/source/talk/libjingle_examples.gyp        | 4 ----
 third_party/webrtc/base/base.gyp                                | 4 ----
 third_party/webrtc/modules/desktop_capture/desktop_capture.gypi | 2 +-
 4 files changed, 1 insertion(+), 13 deletions(-)

diff --git a/third_party/libjingle/source/talk/libjingle.gyp b/third_party/libjingle/source/talk/libjingle.gyp
index 2182561..86c243b 100755
--- a/third_party/libjingle/source/talk/libjingle.gyp
+++ b/third_party/libjingle/source/talk/libjingle.gyp
@@ -633,10 +633,6 @@
               '-lcrypto',
               '-ldl',
               '-lrt',
-              '-lXext',
-              '-lX11',
-              '-lXcomposite',
-              '-lXrender',
               '<!@(<(pkg-config) --libs-only-l nss | sed -e "s/-lssl3//")',
             ],
           },
diff --git a/third_party/libjingle/source/talk/libjingle_examples.gyp b/third_party/libjingle/source/talk/libjingle_examples.gyp
index f69c5dc..e1ff366 100755
--- a/third_party/libjingle/source/talk/libjingle_examples.gyp
+++ b/third_party/libjingle/source/talk/libjingle_examples.gyp
@@ -206,10 +206,6 @@
                 'libraries': [
                   '<!@(pkg-config --libs-only-l glib-2.0 gobject-2.0'
                       ' gthread-2.0 gtk+-2.0)',
-                  '-lX11',
-                  '-lXcomposite',
-                  '-lXext',
-                  '-lXrender',
                 ],
               },
             }],  # OS=="linux"
diff --git a/third_party/webrtc/base/base.gyp b/third_party/webrtc/base/base.gyp
index 330ea82..fa524cc 100644
--- a/third_party/webrtc/base/base.gyp
+++ b/third_party/webrtc/base/base.gyp
@@ -573,10 +573,6 @@
               '-lcrypto',
               '-ldl',
               '-lrt',
-              '-lXext',
-              '-lX11',
-              '-lXcomposite',
-              '-lXrender',
               '<!@(<(pkg-config) --libs-only-l nss | sed -e "s/-lssl3//")',
             ],
           },
diff --git a/third_party/webrtc/modules/desktop_capture/desktop_capture.gypi b/third_party/webrtc/modules/desktop_capture/desktop_capture.gypi
index 6f4a083..79efef8 100644
--- a/third_party/webrtc/modules/desktop_capture/desktop_capture.gypi
+++ b/third_party/webrtc/modules/desktop_capture/desktop_capture.gypi
@@ -95,7 +95,7 @@
             'desktop_capture_differ_sse2',
           ],
         }],
-        ['use_x11 == 1', {
+        ['use_x11 == 1 and <(ozone_platform_wayland) != 1', {
           'link_settings': {
             'libraries': [
               '-lX11',
-- 
1.8.3.2

